tests/integration-tests/tests/networking/test_networking.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import datetime
+import os
+
+import boto3
+import pytest
+
+from assertpy import assert_that
+from cfn_stacks_factory import CfnStack, CfnStacksFactory
+from conftest import _AVAILABILITY_ZONE_OVERRIDES
+
+TIMESTAMP = "-{:%Y%m%d%H%M%S}".format(datetime.datetime.utcnow())
+
+
+@pytest.fixture()
+def networking_stack_factory(request):
+    """Define a fixture to manage the creation and destruction of CloudFormation stacks."""
+    factory = CfnStacksFactory()
+    yield factory
+    if not request.config.getoption("no_delete"):
+        factory.delete_all_stacks()
+
+
+def get_internet_gateway_id(ec2_client, vpc_id):
+    internet_gateway_response = ec2_client.describe_internet_gateways(
+        Filters=[{"Name": "attachment.vpc-id", "Values": [vpc_id]}]
+    )
+    return (
+        internet_gateway_response["InternetGateways"][0]["InternetGatewayId"]
+        if internet_gateway_response["InternetGateways"]
+        else ""
+    )
+
+
+def get_cfn_parameters(availability_zone, internet_gateway_id, public_cidr, vpc_id, private_cidr=None):
+    """Create cloudformation-compatible stack parameter given the variables."""
+    parameters = [
+        {"ParameterKey": "AvailabilityZone", "ParameterValue": availability_zone},
+        {"ParameterKey": "InternetGatewayId", "ParameterValue": internet_gateway_id},
+        {"ParameterKey": "PublicCIDR", "ParameterValue": public_cidr},
+        {"ParameterKey": "VpcId", "ParameterValue": vpc_id},
+    ]
+    if private_cidr:
+        parameters.append({"ParameterKey": "PrivateCIDR", "ParameterValue": private_cidr})
+    return parameters
+
+
+def load_public_template():
```

Comment:
load_public_template and load_public_private_template can become a single function such as:

```
def read_text(path):
    """Read the content of a file."""
    with path.open() as f:
        return f.read()
```